draft-annevk-johannhof-httpbis-cookies.md

```diff
@@ -1193,11 +1216,24 @@ boolean _httpOnlyAllowed_, boolean _allowNonHostOnlyCookieForPublicSuffix_, and
 
 1. Insert _cookie_ into the user agent's cookie store.
 
-1. Remove all expired cookies from the user agent's cookie store.
+1. Return _cookie_.
+
+
+### Garbage Collect Cookies
+
+To **Garbage Collect Cookies** given a newly inserted cookie, _cookie_:
+
+1. Let _expiredCookies_ be a list containing all expired cookies in the user agent's cookie store.
 
-1. Run Remove Excess Cookies for Host given _cookie_'s host.
+1. For each _expiredCookie_ in _expiredCookies_, remove _expiredCookie_ from the user agent's cookie store.
 
-1. Run Remove Global Excess Cookies.
+1. Let _excessHostCookies_ be the result of running Remove Excess Cookies for Host given _cookie_'s host.
+
+1. Let _excessGlobalCookies_ be the result of running Remove Global Excess Cookies.
+
+1. Let _removedCookies_ be the concatenation of _expiredCookies_, _excessHostCookies_, and _excessGlobalCookies_.
+
+1. Return _removedCookies_
```

Comment:
I think we can do a little better. While the caller will have to split up the returned cookies according to the origin/site boundary, they will now do this redundantly for Remove Excess Cookies for Host, right?

I wonder if it would be better if callers simply invoked the three removal algorithms separately.

Reply:
On further reflection I'm okay with trying out this approach. One thing I do think we should change is what this algorithm takes. It shouldn't take a cookie, it should specifically take a host.